app/bundles/EmailBundle/Tests/Controller/EmailListTypeFunctionalTest.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\EmailBundle\Tests\Controller;
+
+use Mautic\CoreBundle\Test\MauticMysqlTestCase;
+use Mautic\EmailBundle\Entity\Email;
+use PHPUnit\Framework\Attributes\DataProvider;
+use Symfony\Component\HttpFoundation\Request;
+
+final class EmailListTypeFunctionalTest extends MauticMysqlTestCase
+{
+    private const PARENT_EMAIL      = 'Parent Email';
+    private const VARIANT_EMAIL     = 'Variant Email';
+    private const TRANSLATION_EMAIL = 'Translation Email';
+
+    /**
+     * @param bool|string|string[] $topLevel
+     * @param string[]             $expectedNames
+     */
+    #[DataProvider('topLevelFilterDataProvider')]
+    public function testGetLookupChoiceListWithTopLevelFilter($topLevel, array $expectedNames): void
```

Comment:
suggestion: Nothing urgent. We can define type of the 1st argument.

```suggestion
    public function testGetLookupChoiceListWithTopLevelFilter(array|string|bool $topLevel, array $expectedNames): void
```